pkg/mount/mountutil.go

```diff
@@ -99,24 +99,47 @@ func (m DiskMountUtil) getDeviceMountAttr(fn getMountData) (DeviceMountAttr, err
 //	returns syspath of that disk from which we can generate ndm given uuid of that disk.
 func getDiskDevPath(partition string) (string, error) {
 	// dev path be like /dev/sda4 we need to remove /dev/ from this string to get sys block path.
-	var diskName string
 	softlink := "/sys/class/block/" + partition
 	link, err := filepath.EvalSymlinks(softlink)
 	if err != nil {
 		return "", err
 	}
-	/*
-		link looks - /sys/devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:0/0:0:0:0/block/sda/sda4
-		parent disk is present after block then partition of that disk
-	*/
-	parts := strings.Split(link, "/")
+
+	parentDisk, ok := getParentBlockDevice(link)
+	if !ok {
+		return "", fmt.Errorf("could not find parent device for %s", link)
+	}
+	return "/dev/" + parentDisk, nil
```

Comment:
Should we also log the parentDisk or diskDevPath for the sysPath?
Perhaps V(3) or V(4)

Reply:
is logged by the caller, but the levelled logs will be done in another PR.